### PR TITLE
Add an initial landing page

### PR DESCRIPTION
**Pull request by justinkrol**
This makes some changes to the project directory structure and component scanning so that we can serve a basic "Hello world" page.

As a follow-up, I'll modify this so that we serve a "student view" page.

### CODE REVIEW

File: src/main/java/app/controllers/HomeController.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package app.controllers;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class HomeController {
+    @GetMapping("/hello")
```
Comment (JFleming4) [inline src/main/java/app/controllers/HomeController.java:8]: should the home route just be /  ?

File: src/main/java/app/deploy/MainConfig.java [hunk 2]
```diff
@@ -1,7 +1,6 @@
-package deploy;
+package app.deploy;
```
Comment (DerekStride) [inline src/main/java/app/deploy/MainConfig.java:1]: Why is this needed?

**Review by JFleming4 — COMMENTED** [review]
Just the one question

File: src/main/java/app/controllers/HomeController.java (hunk 1 above)
Comment (justinkrol) [reply src/main/java/app/controllers/HomeController.java:8]: Yep, this will change in follow up. This PR is more about actually serving something, so I want to be explicit that it's literally a hello world page.

File: src/main/java/app/deploy/MainConfig.java (hunk 2 above)
Comment (justinkrol) [reply src/main/java/app/deploy/MainConfig.java:1]: In `Application.java` above, I've split out the `@SpringBootApplication` annotation because it wasn't finding the controller in its component scan.  

Another option is to pass a list of packages like this:
`@ComponentScan({"deploy", "controllers"})`. 
In that case we would need to include any new packages to the list when they are added.

File: src/main/java/app/controllers/HomeController.java (hunk 1 above)
Comment (JFleming4) [reply src/main/java/app/controllers/HomeController.java:8]: ok makes sense

**Review by DerekStride — APPROVED** [review]

**Review by JFleming4 — APPROVED** [review]

**Review by noahsegal — APPROVED** [review]
